src/west/configuration.py

```diff
@@ -70,19 +75,28 @@ def parse_key(dotted_name: str):
         return section_child
 
     @staticmethod
-    def from_path(path: Path | None) -> '_InternalCF | None':
-        return _InternalCF(path) if path and path.exists() else None
+    def from_paths(paths: list[Path] | None) -> '_InternalCF | None':
+        if not paths:
+            return None
+        paths = [p for p in paths if p.exists()]
+        return _InternalCF(paths) if paths else None
 
-    def __init__(self, path: Path):
-        self.path = path
+    def __init__(self, paths: list[Path]):
         self.cp = _configparser()
-        read_files = self.cp.read(path, encoding='utf-8')
-        if len(read_files) != 1:
-            raise FileNotFoundError(path)
+        self.paths = paths
+        read_files = self.cp.read(self.paths, encoding='utf-8')
+        if len(read_files) != len(self.paths):
+            raise WestNotFound(f"Error while reading one of '{paths}'")
```

Comment:
```suggestion
            raise MalformedConfig(f"Error while reading one of '{paths}'")
```
All `self.paths` have been verified to `.exists()`, so "NotFound" is IMHO misleading.